osu.Game/Graphics/UserInterface/Nub.cs

```diff
@@ -16,49 +11,37 @@
 using osu.Framework.Graphics.Shapes;
 using osu.Framework.Graphics.UserInterface;
 using osu.Game.Overlays;
+using osuTK.Graphics;
 
 namespace osu.Game.Graphics.UserInterface
 {
-    public partial class Nub : Container, IHasCurrentValue<bool>, IHasAccentColour
+    public abstract partial class Nub : Container, IHasCurrentValue<bool>, IHasAccentColour
```

Comment:
I don't get this class. Why is there an abstract base class for nubs if `OsuSliderBar` does not use it? Why is there shared appearance logic in here? Why does `ShearedNub` implement it if the following properties make zero sense for it:

- `Current` (in the original `Nub` it meant if the nub is filled or not, which makes no sense in the sheared implementation as there is no need for an "unfilled" nub there),
- `AccentColour`, `GlowingAccentColour`, `GlowColour` (should always likely be using the colour provider in the sheared case)

Reply:
> I don't get this class. Why is there an abstract base class for nubs if `OsuSliderBar` does not use it? Why is there shared appearance logic in here? Why does `ShearedNub` implement it if the following properties make zero sense for it:
> 
> - `Current` (in the original `Nub` it meant if the nub is filled or not, which makes no sense in the sheared implementation as there is no need for an "unfilled" nub there),
> - `AccentColour`, `GlowingAccentColour`, `GlowColour` (should always likely be using the colour provider in the sheared case)

2. I planned to use sheared toggle-able nubs without the slider in the practice mode ui since a sheared rangeslider and a normal button would've look weird IMO. Hence why I left it in here as an option.

Reply:
> I don't get this class. Why is there an abstract base class for nubs if `OsuSliderBar` does not use it? Why is there shared appearance logic in here? Why does `ShearedNub` implement it if the following properties make zero sense for it:
> 
> - `Current` (in the original `Nub` it meant if the nub is filled or not, which makes no sense in the sheared implementation as there is no need for an "unfilled" nub there),
> - `AccentColour`, `GlowingAccentColour`, `GlowColour` (should always likely be using the colour provider in the sheared case)

I'm confused as to how they aren't being used. The slider uses a nub as it's dragging point so it needs a sheared versión no?

Reply:
The `Nub` class is not referenced anywhere else (like from `OsuSliderBar`), it is only inherited. (This isn't necessarily bad.) Each slider creates its own concrete implementation of the nub but then each of them hooks up the bindables or whatever else on its own. (This isn't necessarily bad either.) So at this point it looks like the only purpose of `Nub` the abstract class is to share visual appearance logic (which I thought was agreed as the wrong way to go).

Reply:
> The `Nub` class is not referenced anywhere else (like from `OsuSliderBar`), it is only inherited. (This isn't necessarily bad.) Each slider creates its own concrete implementation of the nub but then each of them hooks up the bindables or whatever else on its own. (This isn't necessarily bad either.) So at this point it looks like the only purpose of `Nub` the abstract class is to share visual appearance logic (which I thought was agreed as the wrong way to go).

Right, I did apply that logic for the slider but like I really felt this could be done cleanly in the nub because of how much less interwoven the structure is. But it's a 5 minute edit anyways so shall I separate them out ?

Reply:
I think splitting them out is probably for the best.

Reply:
done